### ISSUE TITLE
joblib causing clearml to hang

### ISSUE THREAD

**Issue by marekcygan** [allegroai/clearml#1009]
## Describe the bug

When using joblib with clearml the system hangs.

## To reproduce

I have a minimalistic example of code used to reproduce the bug, it uses joblib, torch and clearml. Code pasted below. The main suspect is auto_connect_frameworks, as when setting auto_connect_frameworks=False the deadlock disappears.

## Expected behaviour

When the joblib needs to fill the cache, the system hangs, clearml is printing lines such as:
```2023-05-03 14:10:06,333 - clearml.model - WARNING - 2 model found when searching for `file:///.../default_140627723917120_2`
2023-05-03 14:10:06,333 - clearml.model - WARNING - Selected model `training` (id=f9570c4759864bf69a08d90ead36c5ea)
2023-05-03 14:10:06,334 - clearml.frameworks - INFO - Found existing registered model id=f9570c4759864bf69a08d90ead36c5ea [/.../default_140627723917120_2] reusing it.
2023-05-03 14:10:08,432 - clearml.frameworks - INFO - Found existing registered model id=db72cca6a8704c71a63f0a9b44f235e2 [/.../default_140627723917120_4] reusing it.
2023-05-03 14:10:10,525 - clearml.frameworks - INFO - Found existing registered model id=db72cca6a8704c71a63f0a9b44f235e2 [/.../default_140627723917120_4] reusing it.
2023-05-03 14:10:13,868 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 [/.../default_140627723917120_3] reusing it.
2023-05-03 14:10:15,996 - clearml.frameworks - INFO - Found existing registered model id=db72cca6a8704c71a63f0a9b44f235e2 [/.../default_140627723917120_4] reusing it.
2023-05-03 14:10:18,040 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 [/.../default_140627723917120_3] reusing it.
2023-05-03 14:10:20,140 - clearml.frameworks - INFO - Found existing registered model id=db72cca6a8704c71a63f0a9b44f235e2 [/.../default_140627723917120_4] reusing it.
2023-05-03 14:10:22,255 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 [/.../default_140627723917120_3] reusing it.
2023-05-03 14:10:24,344 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 [/.../default_140627723917120_3] reusing it.
2023-05-03 14:10:27,513 - clearml.frameworks - INFO - Found existing registered model id=db72cca6a8704c71a63f0a9b44f235e2 [/.../default_140627723917120_4] reusing it.
2023-05-03 14:10:29,624 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 [/.../default_140627723917120_3] reusing it.
2023-05-03 14:10:31,768 - clearml.frameworks - INFO - Found existing registered model id=33a49dddaccc4be7aaf4750f0facdcb1 ```

## Environment
* Server type (self hosted \ app.clear.ml): app.clear.ml
* ClearML SDK Version
* Python Version: 3.9.16
* OS (Windows \ Linux \ Macos) Linux
## Related Discussion

https://clearml.slack.com/archives/CTK20V944/p1683019767054709

**Comment by ainoam** [comment]
Thanks for reporting @marekcygan.

Which clearml package version were you using?

**Comment by burnereugen** [comment]
Hi @marekcygan ! It looks like ClearML thinks that the cached files are models and tries to log them to the backend. I suggest you disable the `joblib` binding and only allow pytorch to log models whose names match a certain wildcard, chosen by you:
```
auto_connect_frameworks={"joblib": False, "pytorch": [""]}  # empty string means no models should be logged
```

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @marekcygan ! ClearML confuses the cached data with models and tries to log the data to the server, that's why the program appears to be hanging. I suggest you disable the `joblib` binding and only allow certain models to be logged by pytorch, via wildcards. See `auto_connect_frameworks` under https://clear.ml/docs/latest/docs/references/sdk/task#taskinit.
Example:

```
auto_connect_frameworks={"joblib": False, "pytorch": [""]}  # empty string = no model should be logged
```

**Comment by marekcygan** [comment]
@eugen-ajechiloae-clearml thank you, works on my end.

What I don't understand however is why `"joblib": False` is not enough?

**Comment by pollfly** [comment]
Hey @marekcygan! Just letting you know that this issue has been resolved in the recently released [v1.11.0](https://github.com/allegroai/clearml/releases/tag/v1.11.0). Let us know if there are any issues :)

**Comment by marekcygan** [comment]
@pollfly thanks!
